### DISCUSSION TITLE
MHIG support

### DISCUSSION THREAD

**Discussion by harman-f** [Mr-MIBonk/M.I.B._More-Incredible-Bash#250] [Show and tell]
Thanks to @lprot we started to go down the road of adding support for MHIG units in M.I.B!!!!
[[@lprot](https://github.com/lprot)](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/commit/67c55ec4adcde2a9e3273a9b8c83d982f35d2855)

Yes, you heard right!

This is by far not finished, main point is currently to NOT break functions for MHI2 units.
Looking not so bad until now...

You are all welcome to help testing

What is working:
- GEM activation during M.I.B install
- own GEM menu structure for MHIG - automatically selected during M.I.B install
- Standard Backup
- VIM
- SVM fix
- GEM will be updated to 4.12 during install
- Flash is in Alpha state - you have to built your own patch folders (same like MHI2) to test this on your unit

Get latest version directly from repo: https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/archive/refs/heads/main.zip

**Please share all MHIG backups you create with M.I.B**

**Comment by harman-f** [comment]
Enabling GEM might/should be working on MHIG now
https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/commit/ab1af9accf6c5a0f11690e84278137ddcc7f024d

Test needed

**Comment by RoDriDC** [comment]
The zip is for MIB High 1? 
![WhatsApp Image 2023-06-15 at 14 41 17](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/97b06bff-a7c0-421c-bd9a-9d585b19eb0f)

**Reply by lprot** [reply]
Yes

**Reply by RoDriDC** [reply]
I have install a previus version. I try to install and make a backup.

**Comment by RoDriDC** [comment]
Update mmi to k1555 firmware and install M.I.B 

![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/1914b047-6d56-4d2e-a817-54263254ca7c)

when install M.I.B de last script dont past 0%.

 Cancel and check de green menu.
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/f83a5e54-eced-43eb-9d41-9b54b24953f7)
Backup!!  : https://mibsolution.one/index.php/f/182181
Backup is no patching firmware 
If you need any more, tell me!!

**Reply by harman-f** [reply]
Did the install activate GEM/Dev mode or had you to do it via OBD?

**Reply by RoDriDC** [reply]
No. GEM need obedeleven to actívate.

**Reply by harman-f** [reply]
Latest test show that GEM can now be activated by M.I.B during install.

Would be great if you could confirm this ;)

Could you upload your backup again? (filehorst.de)
Could not find it on mibsolution.

**Reply by RoDriDC** [reply]
https://filehorst.de/d/evihtleF
Problems: Update firmware k1555, install MIB and latest script does not pass 0%. To access GEM I need to activate the option with OBD. Perhaps not executing the last script correctly is the reason?

**Reply by harman-f** [reply]
0% will be shown for most of the time. Onl at the very end 100% for a short moment.

Could you please share your log file

**Reply by RoDriDC** [reply]
[MU0415-JTB-07409.07.147777076D.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11774616/MU0415-JTB-07409.07.147777076D.log)
This MIB version has enabled the appp uninstall? Maybe I can repeat the whole process to see if it has been my failure. I think 5 minutes passed without the 0%process moving.

**Reply by harman-f** [reply]
Looking at your LOG all - which was already MHIG tested - went well.
FinalScript did run.

Patch files are not yet available ;)
CP/AA of course does not work.

Dev mode should ahve been activated after installing M.I.B.

What do you want to do next?Patch and Update maps?

**Reply by RoDriDC** [reply]
With this mmi unit (mmi plus touch) I don't think there are many more options to be made. When I looked for information for Mib high 1, it was to update the maps I have from 2017 to a current version.

I'm considering getting a mib2 unit and trying to code everything for my car. But first I need to see if I'll be able to do it. XDD

**Reply by harman-f** [reply]
I will look into patching in more detail during the next days.

**Reply by RoDriDC** [reply]
If you need me to try something, let me know.

**Reply by harman-f** [reply]
Could you just run the M.I.B install again with latest version.
You should note one new thing ;)

And share your log file

**Reply by RoDriDC** [reply]
[MU0415-JTB-07409.07.147777076D.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11817945/MU0415-JTB-07409.07.147777076D.log)
Everything works perfect. MIB made a new backup automatically. The log file is in the backup folder. The green screen was displayed and everything finished successfully.

**Reply by harman-f** [reply]
GEM was also updated to 4.12?

**Reply by RoDriDC** [reply]
yes. the new verssion is 4.12s.
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/94b4b2a0-ebdd-435c-a94d-411b1d53d67e)
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/666205e5-6e2e-4b0b-8966-2508a25d2bb8)
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/a159481b-6b34-463e-89f7-6bad67b7e25e)
My unit has VIM enabled and the VIM section shows this information. I don't know if this is relevant to you.

![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/ed41b4f6-315e-4fd4-90d3-ba854ffad8c4)

**Reply by harman-f** [reply]
Can MHIG even play videos?

VIM should work fine. Values look OK.

Many things on MHIG are same or very similar to MHI2.
Just not tested all yet ;)

**Reply by RoDriDC** [reply]
Yes. MHIG play mp4 at 720. 
It is a good unit except for not having AA/CP. I think the rest of the features are identical to MIB2. It has HDD, WLAN, touch, VIM...

**Reply by harman-f** [reply]
Do you have a OBD tool?
Could you give SVM fix a try?

From console it also should work on MHIG. Sadly OBD on my bench is brokern ;)

**Reply by RoDriDC** [reply]
Yeah. I have obdleven. svm fix I did it a couple of days ago when I updated the firmware and it allowed me to clear the error. I have checked with obdleven and now it has the error again. Do you want me to try the M.I.B svm fix or obdeleven?

**Reply by harman-f** [reply]
SVM will come back even after M.I.B install
Would be great if you could confirm once more 😊

**Reply by RoDriDC** [reply]
Error from install MIB.
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/22b084f0-3a5d-4328-a11c-f795d84dff24)
SVM fix fron console.
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/786337bc-38c4-4615-b81e-c320f268cd77)
![imagen](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/33a083de-fd12-4645-ac68-0d1ebbbbb192)

**Reply by RoDriDC** [reply]
Log file svm fix
[MU0415-JTB-07409.07.147777076D.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11829652/MU0415-JTB-07409.07.147777076D.log)

**Reply by harman-f** [reply]
You can delete SVM
Error now via OBD?

**Reply by RoDriDC** [reply]
Once svm fix was executed in the console I was able to clear the error with obdeleven.
The above images are of the process. Error in obdeleven, executed svm fix in console and cleared error in obeyeleven.

**Reply by harman-f** [reply]
Do you by any chance own a Android phone?

**Reply by RoDriDC** [reply]
Yes. redmi note 11 pro.

**Reply by harman-f** [reply]
There seems to be Mirrorlink available on MHIG. 
Will look into that tonight.

**Reply by harman-f** [reply]
Could not get it running on my bench...

**Comment by harman-f** [comment]
What is working now:
- GEM activation during M.I.B install
- own GEM menu structure for MHIG - automatically selected during M.I.B install

**Reply by harman-f** [reply]
Confirmed working:
- SVM fix 
- VIM 
- GEM will be updated to 4.12 during install

**Reply by harman-f** [reply]
Working now (Beta):
- LIVE Patch
- addFec

**Reply by harman-f** [reply]
some more debugging.
Functions are more stable now.
EU Conversion is merged.

Close to release.

**Reply by deermichel** [reply]
qq regarding live patch: so if i have k1555 i don't need any additional patch files? i can just install the current `main`, backup, run aio patch and enjoy? has someone tested that with k1555 yet?

**Reply by harman-f** [reply]
Correct, patch will be generated on unit. 
Patching is safe, in worse case patch will not be applied to unit. Latest test were all successful. 

Addfec might still be a bit buggy, meaning that it might not add a new FEC. But this should be fixed since yesterday. 

Please share your log in any case 😉.

**Reply by harman-f** [reply]
And yes, get latest main

**Reply by RoDriDC** [reply]
It worked on my unit. The only issue was that add Fec didn't work directly from the Fec repository on patch folder. But that was a few days ago and harman-f commented that it should already be solved.
![WhatsApp Image 2023-07-05 at 08 42 42](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/97b1e65f-7228-4960-9476-e58942a09719)
![WhatsApp Image 2023-07-05 at 08 43 12](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/91197355/b30f3b7b-b557-4a27-a661-059780155e75)

**Reply by deermichel** [reply]
Looks good, I haven't tried new maps yet, but the FEC `021000EE` is listed as legal (fw: MHIG_EU_AU_K1555). I did
- install MIB, GEM
- backup (let me know if you want it)
- patch AIO (after auto reboot from green success screen got stuck on a black screen, did a manual 3-button reboot)
- fix SVM (tbh i don't have an OBD interface yet to check it)

Logs attached. Thank you! This type of hardware hacking is super interesting (and terrifying at the same time) :)

[MU0415-JTB-07423.02.158590051D.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/11981585/MU0415-JTB-07423.02.158590051D.log)

![image](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/8472721/28b681ff-ebc8-4bd8-9de8-32f6a7411056)

**Reply by harman-f** [reply]
You can now install latest maps 😊 

How long did you wait on black screen before manual reboot?
Can take up to 60s. Will add that to image soon. 

please upload your backup to mibsolution.

**Reply by deermichel** [reply]
Nice! I think i waited longer than that, and eventually decided to help a bit once it felt safe.

**Reply by Tapisvolant23** [reply]
According to you I have a 2014 audi s3 with MHIG EU AU S1048. I will test but is the unit compatible with the M.I.B?

**Reply by lprot** [reply]
With all MHIG* trains

**Reply by deermichel** [reply]
@harman-f so I tried to update the maps today to the 2023/24 EU version (`P330_N60S5MIBH3_EU`).

However I ran into "No valid update data was found on the media source" and was unable to perform the update (via the MMI Setup/Sysupdate/regular way for doing updates). The update was detected in the red menu, but I cannot start a nav update from there (greyed out). I used a FAT32, 32GB SD card - and my FW is a EU-variant `MHIG_EU_AU_K1555`, also the FEC `021000ee` is still listed as legal. Any idea what's wrong here or how to debug it further?

**Reply by harman-f** [reply]
How does your file/folder structure on SD look like?

**Reply by deermichel** [reply]
copied the 7z contents to the root

![image](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/8472721/a04836a6-9986-462d-86a1-159f44ff1926)

**Reply by harman-f** [reply]
Maps have been working before, just old?

try Navdb delete

**Reply by deermichel** [reply]
ok, the problem was macOS - i reformatted FAT32 with Win & extracted it again, and suddenly the MMI detects the update. I'll install it on my next longer drive (do you know if the nav update can be interrupted?). I wonder if the issue is due to some checksum mismatch with the .DS_Store files that macOS creates.

**Reply by harman-f** [reply]
Yes, MAC does that :)

**Comment by harman-f** [comment]
Continue here:
https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/discussions/276

**Comment by adandashi** [comment]
Can anyone help me find MU0377-MHIG_RW_VW_S1538_1

**Comment by tfojta** [comment]
Patching did not work for me - using MHIG_EU_AU_K1555 firmware. AIO was successful but no new activation key appeared. Tried the FEC txt update. It resulted in Patching failed message and the code shows as illegal.

MMX-flash-done!
--- Reading patch DB...
--- Searching offset and size of root-stage2.ifs...:
[/net/mmx/fs/sda0/apps/sbin/flashlock] Info : Root-Block at Offset (52)!
ifs-root-stage2.ifs offset: 0x00ba0000
ifs-root-stage2.ifs size: 845DCC00
--- OK: OFFSETPART2 in DB and RCC flash match.
--- OK: STAGE2SIZE in DB and RCC flash match.
--- Dumping root-stage2.ifs from RCC flash.
    It will take about 1 minute. Please wait...
--- Patching dumped root-stage2.ifs.
    It will take about 2 minutes. Please wait...
!!! Patching failed.
    Stopping here ...

[MU0415-JTB-07416.04.1477771203.log](https://github.com/user-attachments/files/16400953/MU0415-JTB-07416.04.1477771203.log)
![1](https://github.com/user-attachments/assets/5aa93ad9-0e9a-4f1c-b128-76ebb7068685)
![2](https://github.com/user-attachments/assets/78aa0913-5b7f-4b3d-a12e-a51c7451c626)
![3](https://github.com/user-attachments/assets/ad59f053-9d0c-4ca0-aacd-ec791ffcba5e)

**Comment by bogdanxd103** [comment]
what version can I update my unit , I curently have "MHIG_EU_VW_S1054, MU:0290" and on mib helper it says I can update to "MHIG_EU_VW_K1550, MU:0407" but when I do the update it says that the train is blocked
